property/index.js

```diff
@@ -115,54 +119,133 @@ module.exports = yeoman.generators.Base.extend({
       {
          name: 'defaultValue',
          message: 'Default value[leave blank for none]:',
-         default: null
+         default: null,
+         when: function(answers) {
+          return answers.type !== null && 
+            answers.type !== 'buffer' &&
+            answers.type !== 'any' &&
+            typeChoices.indexOf(answers.type) !== -1;
+        }
       }
     ];
+
     this.prompt(prompts, function(answers) {
+      debug('answers: %j', answers);
       this.name = answers.name || this.name;
       if (answers.type === 'array') {
         var itemType =  answers.customItemType || answers.itemType;
         this.type = itemType ? [itemType] : 'array';
       } else {
         this.type = answers.customType || answers.type;
       }
-      this.required = answers.required;
-      this.defaultValue = answers.defaultValue;
+
+      this.propDefinition = {
+        name: this.name,
+        type: this.type,
+      };
+      if (answers.required) {
+        this.propDefinition = Boolean(answers.required);
```

Comment:
Yeah, that line is what broke some of the behaviour related to comments above... it's supposed to be:
`+      if (answers.required) {
 +        this.propDefinition.required = Boolean(answers.required);`
I initially attempted to avoid the conditional set of the attribute, until I realised it unnecessary forced the occurrence of `require: false` all over the models, when it's certainly better to no bother at all unless is required. So, I'm strongly recommending that we avoid setting required attribute unless is relevant, or gets reset from `true`